### ISSUE TITLE
Custom view starting at current day

### ISSUE THREAD

**Issue by jibbers42** [fullcalendar/fullcalendar#2847]
A forecast view would lock today to the second column and therefore show yesterday, today, and 5 days into the future, no matter what day of the week it is.

It is often nice to be able to review yesterday and also see what is coming up over the next several days. I find that I like week view a lot when it is Monday, but start disliking it more and more the further into the week we get since my view of the future diminishes.

I'm guessing prev and next would need to be disabled in forecast view since that might be confusing - or maybe someone has a better solution for that. Even if disabling is the best option, I don't think that takes anything away from the usefulness of a forecast view.

**Comment by Jibbers42** [comment]
For comparison I have a normal Mon, a normal Thu, and the same Thu in forecast view. I'm not sure about the forecast button placement - maybe it should be right in the group. Maybe it should say "week forecast", but that's getting a bit long. Should it be it's own button, or should it modify the state of week as a toggle (and perhaps month)?

Forecast could be extended to month view too I think, perhaps by locking the current week to the second row, but I'm not sure I see as much value in that view. I guess it could be added later if this forecast concept becomes a thing in fc.

normal week Mon
![week mon](https://cloud.githubusercontent.com/assets/797824/9585217/89b4ad5e-4fe3-11e5-92ee-a6d8dac02c79.png)

normal week Thu
![week thu](https://cloud.githubusercontent.com/assets/797824/9585230/978d895a-4fe3-11e5-8b86-a8087cf9af7e.png)

new forecast Thu
![forecast thu](https://cloud.githubusercontent.com/assets/797824/9585249/b0e4b4c8-4fe3-11e5-95ac-dd8721430c85.png)

**Comment by espen** [comment]
Ok, that's seems handy. Good suggestion.

btw: it is possible to create your own views using http://fullcalendar.io/docs/views/Custom_Views/

**Comment by arshaw** [comment]
this seems useful, but the way in which Custom Views are architected, it detects the interval we are dealing with (in this case "week") and makes the first day the start-of-whatever. Providing options to override this behavior seems to be the way to go. Not exactly user what the API/options should look like tho. would require more thought.

**Comment by kenzan100** [comment]
I was searching for the same solution, and below workaround matched my need.
you could throw a moment.js weekday value to firstDay parameter when initializing your fullcalendar.
```
$('#yourFullCalendar').fullCalendar({
  ...
  firstDay = moment().format('e');
});
```

**Comment by arshaw** [comment]
Here's how I think the API should differentiate between the two cases. If the user *wants* the date to always start at the beginning of the week, they should use the stock `agendaView` or for more advance uses, should see #3113.

However, if they DON'T want to start at the beginning of the week (like what @Jibbers42 desires), then a custom view is necessary:

```
views: {
    agendaSevenDays: {
        duration: { days: 7 }
    }
}
```

If you were to leverage this currently, you'd end up w/ the undesirable start-of-week. It doesn't differentiate between 7 days VS 1 week. The innovation of this ticket will be treating `{ weeks: 1 }` different from `{ days: 7 }`

**Comment by arshaw** [comment]
I'm broadening this task to include other views. `listYear` has the same problem. @illuminate3 please give a thumbsup to the top comment on this page.

**Comment by illuminate3** [comment]
Thanks!

The ListYear view is a really nice feature but having it start from January was a little strange because all the other functionality can start with today which I thought was a tad odd.

**Comment by taddeozanga** [comment]
Thank you so much for your control! it rocks! I have the  same problem and tried everything and I was expecting the behaviour of days: 7 to be different from week but I discovered it is the same as you are mentioning.... When can we have this implemented? This is something I really need to make the control work for me.... are there any current workarounds? I tried to play with the firstday parameter but that is unreliable... THANK YOU!!!
